### ISSUE TITLE
There's no documentation on how to add project dependencies

### ISSUE THREAD

**Issue by elijahmontenegro** [unitejs/cli#7]
Hello,
I attempted to create a Polymer 3 app using PaperElements web components, I went ahead and
installed a couple elements (with some guidance from the PolymerElements documentation) the usual way by using "npm install" doing that doesn't work after testing with the "**gulp build / serve commands**". I then tried using the unite command line --i.e: "unite clientPackage --opetation=add --packageName=@polymer/paper-toolbar", however the same applied. There is no documentation on how to get this kind of package and presumably it's own dependencies added to the build in a graceful and effective manner.
I then realized that the unite commands and documentation were going to be no help. 

There needs to be documentation for how to add this kind of package, and importantly, it's dependencies, in a graceful way. In other words if the cli or documentation does not cover this then it should be added right away. 

**Issues Encounted**
In summary the problems I've faced in my testing are below:
- There's no documentation on how to add project dependencies
- There's no specific command for adding those kinds of project dependencies like PolymerElements ( or in the case of React, it might be @material-ui/core for example) (again if there could be clarification / additions to the docs about this that would be a huge help!)
- Adding dependencies using "unite clientPackage" should also find and add those sub-dependencies, for example, if adding '@polymer/paper-toolbar', that in it-self has it's own sub-dependencies, and so those should be included as well to the unite.json and,or referenced in the dist build itself. 
Hense a user will see this in their served website's console,  this is after installing their intended package (in this example I use @polymer/paper-toolbar). Below are errors for the missing dependencies that paper-toolbar uses. Its worth noting that these sub-dependencies are located in the project.json within the @polymer/paper-toolbar folder but aren't detected  by the unite clientPackage --add command
```
:9000/node_modules/@polymer-transpiled/polymer/polymer-legacy.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:9000/@polymer/paper-styles/default-theme.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:9000/@polymer/paper-styles/typography.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:9000/@polymer/iron-flex-layout/iron-flex-layout.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:9000/node_modules/@polymer-transpiled/polymer/lib/utils/html-tag.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
(index):1 Uncaught (in promise) Error: Fetch error: 404 Not Found
  Instantiating http://localhost:9000/node_modules/@polymer-transpiled/polymer/polymer-legacy.js
  Loading http://localhost:9000/node_modules/@polymer-transpiled/paper-toolbar//paper-toolbar.js
  Loading dist/entryPoint
    at fetch.js:37
```
- Lastly, the build process seems to be confused when adding packages like the paper-toolbar example below (look in the Additional Information below)

**Steps to Replicate**
Step 1
Establish a unitejs project using the below
`unite configure --packageName=xtremeqatool-unitejs-react-ts_issuerepro --profile=PolymerTypeScript --linter=None --unitTestRunner=None --e2eTestRunner=None --cssPost=None --cssLinter=None --documenter=None --packageManager=Npm --title=Title --shortName="Short Name" --description=Description --keywords=Keywords --organization=Organization --copyright=Copyright --webSite="Web Site" --namespace=Namespace --author=Author --authorEmail=Email --authorWebSite="Author Web Site" --outputDirectory=xtremeqatool-unitejs-react-ts_issuerepro`
Next in the Bash terminal, 'cd www' and run 'npm install'. After that, execute 'gulp theme-build'.

Step 2
Add any element from https://www.webcomponents.org/author/PolymerElements
--e.g: `unite clientPackage --operation=add --packageName=@polymer/paper-toolbar`

Step 3
Modify the Child.ts to include the required import for the element to be used in the html file
```
/**
 * Child class.
 *
 * @export
 * @class Child
 */
import { customElement, property } from "@polymer/decorators/src/decorators";
import { PolymerElement, html } from "@polymer/polymer/polymer-element";
import "@polymer/paper-toolbar/paper-toolbar.js"; // Add the element here according to the polymer element instructions
...
```

Step 3
Modify the Child.html to be something about the same as below
```
<span class="child">
    <paper-toolbar>
        <h2>{{message}}</h2>
    </paper-toolbar>
</span>
```

Step 4
In Bash, execute 'gulp build', and then after that, execute 'gulp serve'.

Step 5 
In the website served, hit [F12] and observe the error in the console.
```
:9000/node_modules/@polymer/paper-toolbar/paper-toolbar/paper-toolbar.js:1 Failed to load resource: the server responded with a status of 404 (Not Found)
(index):1 Uncaught (in promise) Error: Fetch error: 404 Not Found
  Instantiating http://localhost:9000/node_modules/@polymer/paper-toolbar/paper-toolbar/paper-toolbar.js
  Loading http://localhost:9000/dist/child/child.js
  Loading dist/entryPoint
    at fetch.js:37
```
**Additional Information**
I noticed at step 5 it was referencing 'paper-toolbar/paper-toolbar/paper-toolbar' instead of 'paper-toolbar/paper-toolbar', so in 'app-module-config.js' I corrected the reference manually.
```
"@polymer/paper-toolbar": "node_modules/@polymer/paper-toolbar/paper-toolbar"
to
"@polymer/paper-toolbar": "node_modules/@polymer/paper-toolbar"
```
 I then was getting this error message in the console: 
EDIT: the error was bypassed by using a transpile alias, however like mentioned in the **Issues Encounted**, I was then missing paper-toolbar's dependencies which is a thing i want to avoid having to add manually to the unite.json every time I add a different PolymerElement.
```
(index):1 Uncaught (in promise) Error: Unable to dynamically transpile ES module
   A loader plugin needs to be configured via `SystemJS.config({ transpiler: 'transpiler-module' })`.
  Instantiating http://localhost:9000/node_modules/@polymer/paper-toolbar/paper-toolbar.js
  Loading http://localhost:9000/dist/child/child.js
  Loading dist/entryPoint
    at transpile (instantiate.js:470)
    at instantiate.js:245
```

Resources used
https://unitejs.com/
https://www.webcomponents.org/author/PolymerElements
https://www.polymer-project.org/3.0/start/

**Comment by elijahmontenegro** [comment]
Apologies, I accidentally closed this, please still look into this issue. 

-EM

**Comment by obany** [comment]
Thanks for the comprehensive writeup.

This was a little more involved than I had hoped but in order to facilitate what you were trying to do I need to upgrade the polymer framework to the release version, which in turn meant a change in the way transpilation was done. A new release v1.8.1 is now out.

The steps to include the element you wanted is as follows.

Add the client packages, these have to be done manually as there is no way to guess which of the components are required by the others in an easy fashion. Also the font-roboto package supplies a default main as roboto.js which then breaks its inclusion by one of the other elements, hence the main=*

```shell
unite clientPackage --operation=add --packageName=@polymer/paper-toolbar
unite clientPackage --operation=add --packageName=@polymer/paper-styles
unite clientPackage --operation=add --packageName=@polymer/iron-flex-layout
unite clientPackage --operation=add --packageName=@polymer/font-roboto --main=*
```

Include the component in the source, note we are including the module not the source itself.

```ts
import "@polymer/paper-toolbar";
```

And update the template
```html
<span class="child">
    <paper-toolbar>
        <h2>{{message}}</h2>
    </paper-toolbar>
</span>
```

You should then successfully be able to run gulp build and serve without problem.

I will flag this as needs more documentation.

**Comment by elijahmontenegro** [comment]
Verified the issues mentioned are fixed with the latest unitejs-cli (v1.8.1) but will leave the issue open for more documentation to be added. 

All the best, 
-EM
